ReactSkia/views/common/RSkImageCacheManager.cpp

```diff
@@ -0,0 +1,146 @@
+
+  void setCpuImageCacheLimit(size_t cacheLimit) {
+    SkGraphics::SetResourceCacheTotalByteLimit(cacheLimit);
+    cpuCacheLimit_ = SkGraphics::GetResourceCacheTotalByteLimit();
+  }
+
+#ifdef RNS_SHELL_HAS_GPU_SUPPORT
+  void setGpuImageCacheLimit(size_t cacheLimit) {
+    GrDirectContext* gpuContext = RSkSurfaceWindow::getDirectContext();
+    if(gpuContext) {
+      gpuContext->setResourceCacheLimit(cacheLimit);  
+      gpuCacheLimit_ = gpuContext->getResourceCacheLimit();
+    }
+  }
+#endif
+
+  sk_sp<SkImage> makeImageData(const char *path) {
+    sk_sp<SkData> data = SkData::MakeFromFileName(path);
+    if (!data) {
+      RNS_LOG_ERROR("Unable to make SkData for path :" << path);
+      return nullptr;
+    }
+    return( SkImage::MakeFromEncoded(data ));
+  }
+  
+  bool evictAsNeeded() {
+
+  	size_t cpuCacheUsed{0} , gpuCacheUsed{0};
+    int fOldCount{0}, evictCount{0};
+    map<size_t, sk_sp<SkImage>>::iterator it;
+
+/*Read memory usage on CPU/GU Caches*/
+  #ifdef RNS_SHELL_HAS_GPU_SUPPORT
+    GrDirectContext* gpuContext =RSkSurfaceWindow::getDirectContext();
+    if(gpuContext) {
+      gpuContext->getResourceCacheUsage(&fOldCount, &gpuCacheUsed);       
+    }
+  #endif
+    cpuCacheUsed = SkGraphics::GetResourceCacheTotalBytesUsed();
+    RNS_LOG_DEBUG("CPU CACHE consumed bytes: "<<cpuCacheUsed<< ",, GPU CACHE consumed bytes: "<<gpuCacheUsed);
+
+    /*Evict entry on reaching HWM. This logic to be enchanced based on growing need and clarity.*/
+    for(it = ImageCacheMap.begin();(it != ImageCacheMap.end()) && (evictCount < EVICT_COUNT);) {
+      if((cpuCacheUsed < SKIA_CPU_IMAGE_CACHE_HWM_LIMIT) && ( gpuCacheUsed < SKIA_GPU_IMAGE_CACHE_HWM_LIMIT))
+        break;
+      if((it->second)->unique()) {
+          ImageCacheMap.erase(it++);
+          evictCount++;
+          RNS_LOG_DEBUG("Evicting Entry from RNS Image Hash Map ...");
+      } else {
+          ++it;
+      }
+    }
+    return ((cpuCacheUsed < SKIA_CPU_IMAGE_CACHE_LIMIT) && ( gpuCacheUsed < SKIA_GPU_IMAGE_CACHE_LIMIT));
+  }
+
+}//namespace
+
+sk_sp<SkImage> getImageData(const char *path) {
+
+  hash<string> hashedKey; 
+  sk_sp<SkImage> imageData{nullptr};
+  /*Set Cache limit, if not set before*/
+  if(cpuCacheLimit_ != SKIA_CPU_IMAGE_CACHE_LIMIT) {
+    setCpuImageCacheLimit(SKIA_CPU_IMAGE_CACHE_LIMIT);
+  }
+#ifdef RNS_SHELL_HAS_GPU_SUPPORT
+  if(gpuCacheLimit_ != SKIA_GPU_IMAGE_CACHE_LIMIT) {
+    setGpuImageCacheLimit(SKIA_GPU_IMAGE_CACHE_LIMIT);
+  }
+#endif
+  if(!path) {
+    RNS_LOG_ERROR("Invalid File");
+    return nullptr;
+  }
+ /*first check file entry in hash map, if entry not exist, create imageData*/
+  map<size_t, sk_sp<SkImage>>::iterator it = ImageCacheMap.find(hashedKey(path));
+  imageData= ((it != ImageCacheMap.end()) ? it->second : nullptr);
+  if(!imageData) {
+    imageData = makeImageData(path);
+    /*Add entry to hash map only if the cache mem usage is with in the limit*/
+    if(evictAsNeeded() && imageData) {
```

Comment:
we may need to ensure `getImageData` is executed in the same thread. otherwise, to have a lock for the internal ImageCacheMap.

besides, i am thinking if `getImageData` doing too much stuffs. we just do both map lookup and map iterate(inside `evictAsNeeded`). we may later improve the cache design by somehow like a soft eviction in worker thread.
anyway, i agree this is an intermediate simple cache design and fine with this. cc @munezbn if you have some comment.